src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MAGNIFICATION_FACTOR = 1.5;
+    private int size = 0;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= items.length - 1) {
+            resize((int) (items.length * MAGNIFICATION_FACTOR));
+        }
+        items[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index " + index + " non-existent");
+        }
```

Comment:
Extract private method
```suggestion
        checkIndexForAdd()
```